finality-aleph/src/party/impls.rs

```diff
@@ -0,0 +1,366 @@
+use std::{collections::HashSet, fmt::Debug, marker::PhantomData, sync::Arc, time::Duration};
+
+use aleph_bft::{DelayConfig, SpawnHandle};
+use aleph_primitives::KEY_TYPE;
+use async_trait::async_trait;
+use futures::channel::oneshot;
+use log::{debug, trace, warn};
+use sc_client_api::Backend;
+use sp_consensus::SelectChain;
+use sp_keystore::CryptoStore;
+use sp_runtime::traits::{Block as BlockT, Header, NumberFor, SaturatedConversion};
+
+use crate::{
+    crypto::{AuthorityPen, AuthorityVerifier, Keychain},
+    data_io::{ChainTracker, DataStore, OrderedDataInterpreter},
+    default_aleph_config, mpsc,
+    network::{split, ManagerError, RequestBlocks, SessionManager},
+    party::{
+        aggregator,
+        authority::{SubtaskCommon, Subtasks, Task as AuthorityTask},
+        backup::ABFTBackup,
+        traits::{Block, ChainState, NodeSessionManager, SessionInfo},
+    },
+    AuthorityId, ClientForAleph, JustificationNotification, Metrics, NodeIndex, SessionBoundaries,
+    SessionId, SessionPeriod, SplitData, UnitCreationDelay,
+};
+
+pub struct NodeSessionManagerImpl<C, SC, B, RB, BE>
```

Comment:
Maybe we could throw this together with `aggregator.rs`, `member.rs`, `chain_tracker.rs` and `data_store.rs` inside one folder named something like `manager`/`session_manager`/`node_manager`? Right now all those files are thrown in party and are used only by this NodeSessionManager. They don't need to be public to the whole crate and in my opinion this would organize this folder a little bit more.

Reply:
ok, done